### PR TITLE
feat: navigate to widget-selected token

### PR DESCRIPTION
**Pull request by zzmp**
Updates the Token Details page to match the token selected in the widget, if the current page's token is no longer displayed.

NB: This has a visual bug where navigating to the new token goes back to a LoadingTokenDetails state. <s>I'm working on fixing this, but it will likely be a follow-up (unless it is a very small change).</s> This is fixed in a follow-up: #4981.

See WEB-1467

### CODE REVIEW

File: src/components/Widget/inputs.tsx [hunk 1]
```diff
@@ -34,8 +34,9 @@ export function useSyncWidgetInputs(defaultToken?: Currency) {
   useEffect(() => {
     // Avoid overwriting tokens if none are specified, so that a loading token does not cause layout flashing.
     if (!defaultToken) return
-    setTokens({
-      [Field.OUTPUT]: defaultToken,
+    setTokens((tokens) => {
+      if (Object.values(tokens).some((token) => token?.equals(defaultToken))) return tokens
+      return { [Field.OUTPUT]: defaultToken }
```
Comment (aballerr) [inline src/components/Widget/inputs.tsx:39]: could use a ternary here

File: src/pages/TokenDetails/index.tsx [hunk 2]
```diff
@@ -91,28 +92,34 @@ export default function TokenDetails() {
...
       }
     },
     [isNative, navigate, tokenQueryData?.project?.tokens]
   )
-  useOnGlobalChainSwitch(switchChains)
+  useOnGlobalChainSwitch(navigateToTokenForChain)
+  const navigateToWidgetSelectedToken = useCallback(
+    (input: Currency | undefined, output: Currency | undefined) => {
+      const update = output || input
+      if (!token || input?.equals(token) || output?.equals(token) || !update) return
```
Comment (aballerr) [inline src/pages/TokenDetails/index.tsx:111]: ```input?.equals(token) || output?.equals(token)```

I think these might be redundant with the !update at the end

File: src/components/Widget/index.tsx [hunk 3]
```diff
@@ -36,22 +36,29 @@ const WIDGET_ROUTER_URL = 'https://api.uniswap.org/v1/'
...
+export default function Widget({ defaultToken, onTokensChange, onReviewSwapClick }: WidgetProps) {
   const locale = useActiveLocale()
   const theme = useIsDarkMode() ? DARK_THEME : LIGHT_THEME
   const { connector, provider } = useWeb3React()
 
   const { inputs, tokenSelector } = useSyncWidgetInputs(defaultToken)
   const { settings } = useSyncWidgetSettings()
   const { transactions } = useSyncWidgetTransactions()
+  const onSwitchChain = useCallback(
+    // TODO: Widget should not break if this rejects - upstream the catch to ignore it.
```
Comment (aballerr) [inline src/components/Widget/index.tsx:52]: do we have a task for this, just want to make sure we don't forget to do this follow up

Comment (zzmp) [reply src/components/Widget/index.tsx:52]: https://uniswaplabs.atlassian.net/browse/WEB-1757

File: src/pages/TokenDetails/index.tsx (hunk 2 above)
Comment (zzmp) [reply src/pages/TokenDetails/index.tsx:111]: Not redundant - but I rearranged the conditionals in case it makes it clearer.

**Review by aballerr — APPROVED** [review]

**Pull request merged**